src/write_http.c

```diff
@@ -540,6 +539,22 @@ static int config_set_format (wh_callback_t *cb, /* {{{ */
         return (0);
 } /* }}} int config_set_format */
 
+static int wh_config_append_string (const char *name, struct curl_slist **dest, /* {{{ */
+    oconfig_item_t *ci)
+{
+  if ((ci->values_num != 1) || (ci->values[0].type != OCONFIG_TYPE_STRING))
+  {
+    WARNING ("write_http plugin: `%s' needs exactly one string argument.", name);
+    return (-1);
+  }
+
+  *dest = curl_slist_append(*dest, ci->values[0].value.string);
```

Comment:
**curl_slist_append** is tricky. Here *dest will leak if it returns NULL since we clobbered it and lost the change of freeing it. I'd use a temp variable here to store and check the return value, and then overwrite *dest. (The original http_write code has these leaks as well.)